src/com/MPISinglePortsCommunication.hpp

```diff
@@ -76,8 +77,14 @@ class MPISinglePortsCommunication : public MPICommunication {
 
   std::string _addressDirectory;
 
-  /// Remote rank -> communicator map
-  std::map<int, MPI_Comm> _communicators;
+  /// A map of direct communication channels based on MPI_COMM_SELF on both sides
+  std::map<int, MPI_Comm> _direct;
+
+  /// The global intercommunicator that connects all ranks
+  MPI_Comm _global = MPI_COMM_NULL;
```

Comment:
Why again do we need `_direct` **and** `_global` ?

Reply:
The `_direct` communication covers the case where P2P communication is required, namely:

* Master-Salves communication
* Master-Master communication

Once `acceptConnectionAsServer()` and `requestConnectionAsClient()` are whilst establishing the client-server (aka p2p, m2n) communication, this class connects both the client and the server communicator.
Once that happened, this `_global` communicator can be used for arbitrary p2p communication as well as collectives.

Reply:
:+1: I did not fully get this from the comments above though. Could you please add this?